src_cpp/cppBridge/NerlNIF.cpp

```diff
@@ -81,6 +81,71 @@ static ERL_NIF_TERM cppBridgeControllerDeleteModel_nif(ErlNifEnv* env, int argc,
 }
 //----------------------------------------------------------------------------------------------------------------------
 
+// ----- Weights functions -----
+
+// Get weights
+static ERL_NIF_TERM get_weights_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+{
+    std::vector<double> weights_list, bias_list, combined_list;
+    MatrixXd weights_mat;
+    VectorXd bias;
+    int mid; // TODO add it
+    std::vector<std::shared_ptr<ANN::Weights>> vec_of_weights_ptrs;
+
+    // Get the singleton instance
+    cppBridgeController *s = s->GetInstance();
+
+    // Get the model from the singleton
+    //std::shared_ptr<SANN::Model> modelPtr = s-> getModelPtr(mid);
+
+    // Get the weights_list ptr
+    //vec_of_weights_ptrs = modelPtr->get_weights_of_model();
+
+    //for(int i = 0; ){
```

Comment:
Get rid of commented lines as much as possible. You have git to pick at the history or even restore it.